src/main/java/com/fasterxml/jackson/databind/ser/std/NumberSerializer.java

```diff
@@ -55,6 +73,21 @@ public JsonSerializer<?> createContextual(SerializerProvider prov,
         if (format != null) {
             switch (format.getShape()) {
             case STRING:
+                if (format.hasPattern()) {
```

Comment:
```suggestion
                // [databind#1114] since 2.17 : Support @JsonFormat for String, numbers, using String.format()
                if (format.hasPattern()) {
```